### ISSUE TITLE
[VsCode] Connecting to a task hub results in error like 'npm list -g --depth=0 npm ERR! ...'

### ISSUE THREAD

**Issue by scale-tone** [microsoft/DurableFunctionsMonitor#21]
Starting from v5.2 DfMon's VsCode extension [resolves the full true path to func.exe](https://github.com/microsoft/DurableFunctionsMonitor/blob/main/durablefunctionsmonitor-vscodeext/src/BackendProcess.ts#L267) by executing `npm list -g --depth=0` (to get the location of npm's global packages) and then looking for azure-functions-core-tools package in that folder.

It appears, that `npm list -g --depth=0` might fail, if there's any corruption/incompatibility in those global packages.

If you're getting an error related to this, then the quick solution is to explicitly specify full path to `func.exe` (or just `func` on some platforms) via this setting:

![image](https://user-images.githubusercontent.com/5447190/156617218-d35ddc6c-f8d7-45a9-89dd-086d5a4cdf30.png)

On Windows that path typically looks like `C:\Users\username\AppData\Roaming\npm\node_modules\azure-functions-core-tools\bin\func.exe`. On other platforms it will be different, but similarly related to npm's global folder.

**Comment by mikeleuer** [comment]
Also getting this error 
![image](https://user-images.githubusercontent.com/97471415/156618105-8496e19e-a7d0-47ba-81e2-e716fdea1d28.png)

**Comment by scale-tone** [comment]
@mikeleuer , do you have [azure-functions-core-tools](https://www.npmjs.com/package/azure-functions-core-tools) **globally** installed (since it is a requirement for VsCode ext to work) ?

**Comment by scale-tone** [comment]
@mikeleuer , in that case the solution for you should be the same - just specify the full path to func.exe (wherever it is on your system) via `Custom Path to Azure Functions Host` setting.

**Comment by mikeleuer** [comment]
@scale-tone That worked thank you. "C:\Program Files\Microsoft\Azure Functions Core Tools\func.exe" was the location of the file. Hopefully the extension will auto detect the location with updates.

**Comment by junalmeida** [comment]
Would you consider to fall back to system path "func"? I use `yarn` as my package manager.

**Comment by scale-tone** [comment]
@junalmeida , sure, the fix is on its way. Expect a newer version in the next few days. 
And I will consider checking yarn global package folder as well.

**Comment by scale-tone** [comment]
fixed in v5.3

**Comment by junalmeida** [comment]
It is not working for me, this is the output I get when it tries to start the backend:

``` 
npm list -g failed. Command failed: npm list -g --depth=0
npm ERR! peer dep missing: typescript@>=2.3.0-dev || >=2.4.0-dev || >=2.5.0-dev || >=2.6.0-dev || >=2.7.0-dev || >=2.8.0-dev || >=2.9.0-dev || >=3.0.0-dev || >= 3.1.0-dev || >= 3.2.0-dev, required by tslint@5.20.1
npm ERR! missing: loose-envify@^1.1.0, required by react@17.0.2
npm ERR! missing: object-assign@^4.1.1, required by react@17.0.2
npm ERR! missing: loose-envify@^1.1.0, required by react-dom@17.0.2
npm ERR! missing: object-assign@^4.1.1, required by react-dom@17.0.2
npm ERR! missing: scheduler@^0.20.2, required by react-dom@17.0.2
npm ERR! peer dep missing: typescript@>=2.1.0 || >=2.1.0-dev || >=2.2.0-dev || >=2.3.0-dev || >=2.4.0-dev || >=2.5.0-dev || >=2.6.0-dev || >=2.7.0-dev || >=2.8.0-dev || >=2.9.0-dev || >= 3.0.0-dev || >= 3.1.0-dev, required by tsutils@2.29.0
Using Functions host: C:\Users\junior\AppData\Local\Yarn\bin\func
Attempting to start the backend from c:\Users\junior\.vscode\extensions\durablefunctionsmonitor.durablefunctionsmonitor-5.3.0\backend on http://localhost:37072/a/p/i...
```
`No response within 60 seconds. Ensure you have the latest Azure Functions Core Tools installed globally.`

It seems the file it is trying to start is not correct, as `bin\func` is for bash on *nix. On windows it should be either `bin\func.cmd` or `C:\Users\junior\AppData\Local\Yarn\Data\global\node_modules\azure-functions-core-tools\bin\func.exe`.

**Comment by scale-tone** [comment]
Apologize, I should have tested this with yarn on Windows (I only tested on Linux).

Please, specify the explicit path to func.exe as a workaround.

**Comment by junalmeida** [comment]
No need to apologize, this happens. And yes I can confirm it works if I set `func.exe` explicit.

**Comment by scale-tone** [comment]
Re-fixed in v5.4. Please, validate
